analysis/monkwindwalker/src/modules/covenants/CoordinatedOffensive.tsx

```diff
@@ -0,0 +1,116 @@
+import { formatNumber, formatPercentage } from 'common/format';
+import SPELLS from 'common/SPELLS';
+import Analyzer, { Options, SELECTED_PLAYER, SELECTED_PLAYER_PET } from 'parser/core/Analyzer';
+import calculateEffectiveDamage from 'parser/core/calculateEffectiveDamage';
+import conduitScaling from 'parser/core/conduitScaling';
+import Events, { SummonEvent, CastEvent, DamageEvent } from 'parser/core/Events';
+import BoringSpellValueText from 'parser/ui/BoringSpellValueText';
+import ItemDamageDone from 'parser/ui/ItemDamageDone';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
+
+import { ABILITIES_CLONED_BY_SEF } from '../../constants';
+
+class CoordinatedOffensive extends Analyzer {
+  CO_MOD = 0;
+  totalDamage = 0;
+  CO_Active: boolean = false;
+  cloneIDs = new Set();
+  cloneMap: Map<number, number> = new Map<number, number>();
+
+  constructor(options: Options) {
+    super(options);
+    const conduitRank = this.selectedCombatant.conduitRankBySpellID(
+      SPELLS.COORDINATED_OFFENSIVE.id,
+    );
+    if (!conduitRank) {
+      this.active = false;
+      return;
+    }
+
+    this.CO_MOD = conduitScaling(0.088, conduitRank);
+
+    //summon events (need to track this to get melees)
+    this.addEventListener(
+      Events.cast.by(SELECTED_PLAYER).spell(SPELLS.STORM_EARTH_AND_FIRE_CAST),
+      this.CO_Deactivator,
+    );
+    this.addEventListener(
+      Events.cast.by(SELECTED_PLAYER).spell(SPELLS.STORM_EARTH_AND_FIRE_FIXATE),
+      this.CO_Activator,
+    );
+    this.addEventListener(
+      Events.summon
+        .by(SELECTED_PLAYER)
+        .spell([SPELLS.STORM_EARTH_AND_FIRE_EARTH_SPIRIT, SPELLS.STORM_EARTH_AND_FIRE_FIRE_SPIRIT]),
+      this.trackSummons,
+    );
+    this.addEventListener(
+      Events.damage.by(SELECTED_PLAYER_PET).spell(SPELLS.MELEE),
+      this.handleMelee,
+    );
+    this.addEventListener(
+      Events.damage.by(SELECTED_PLAYER_PET).spell(ABILITIES_CLONED_BY_SEF),
```

Comment:
should this also trigger on damage from the player while SEF is active?

Reply:
Nope, CO only buffs dmg from the 2 clones that are not the player